bee-api/bee-rest-api/src/endpoints/mod.rs

```diff
@@ -10,42 +10,102 @@ pub mod rejection;
 pub mod routes;
 pub mod storage;
 
-use std::{any::TypeId, convert::Infallible};
+use std::{any::TypeId, convert::Infallible, ops::Deref, sync::Arc};
 
 use async_trait::async_trait;
-use bee_gossip::NetworkCommandSender;
-use bee_ledger::workers::consensus::ConsensusWorker;
+use bee_gossip::{Keypair, NetworkCommandSender, PeerId};
+use bee_ledger::workers::consensus::{ConsensusWorker, ConsensusWorkerCommand};
 use bee_protocol::workers::{
-    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, PeerManager, PeerManagerResWorker,
-    RequestedMessages,
+    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, MessageSubmitterWorkerEvent, PeerManager,
+    PeerManagerResWorker, RequestedMessages,
 };
 use bee_runtime::{
-    node::{Node, NodeBuilder},
+    event::Bus,
+    node::{Node, NodeBuilder, NodeInfo},
+    resource::ResourceHandle,
     worker::{Error as WorkerError, Worker},
 };
 use bee_tangle::{Tangle, TangleWorker};
+use config::RestApiConfig;
 use log::{error, info};
+use rejection::CustomRejection;
+use storage::StorageBackend;
+use tokio::sync::mpsc;
 use warp::{http::StatusCode, Filter, Rejection, Reply};
 
-use self::{config::RestApiConfig, rejection::CustomRejection, storage::StorageBackend};
 use crate::types::body::{DefaultErrorResponse, ErrorBody};
 
 pub(crate) type NetworkId = (String, u64);
 pub(crate) type Bech32Hrp = String;
+#[cfg(feature = "dashboard")]
+pub(crate) type DashboardUsername = String;
 
 pub(crate) const CONFIRMED_THRESHOLD: u32 = 5;
 
-pub fn init_full_node<N: Node>(
-    rest_api_config: RestApiConfig,
-    protocol_config: ProtocolConfig,
-    network_id: NetworkId,
-    bech32_hrp: Bech32Hrp,
-    node_builder: N::Builder,
-) -> N::Builder
+pub fn init_full_node<N: Node>(init_config: InitConfigFullNode, node_builder: N::Builder) -> N::Builder
 where
     N::Backend: StorageBackend,
 {
-    node_builder.with_worker_cfg::<ApiWorkerFullNode>((rest_api_config, protocol_config, network_id, bech32_hrp))
+    node_builder.with_worker_cfg::<ApiWorkerFullNode>(init_config)
+}
+
+pub fn init_entry_node<N: Node>(init_config: InitConfigEntryNode, node_builder: N::Builder) -> N::Builder
```

Comment:
this function doesn't require documentation, so I guess, it's a crate-internal function.  Can we therefore reduce its visibility to `pub(crate)`?

Reply:
just checked `lib.rs`. Why are we not warning about missing docs?

Reply:
> just checked lib.rs. Why are we not warning about missing docs?
that´s a good question and not optimal! i think this however should be done in a separate PR do you agree? @Alex6323

Reply:
> this function doesn't require documentation, so I guess, it's a crate-internal function. Can we therefore reduce its visibility to pub(crate)?

the node builder makes use of the functions so the must be public

Reply:
right, I realized that when I saw that the `deny(missing_docs)` was disabled in `lib.rs`.

Reply:
Can you enable it again, and fix the docs?

Reply:
As discussed in Discord, this should be handled in a separate PR. I created an issue for it: https://github.com/iotaledger/bee/issues/1277